shared/metrics/src/test/java/io/pravega/shared/MetricsNamesTest.java

```diff
@@ -17,19 +17,42 @@
 @Slf4j
 public class MetricsNamesTest {
 
-  @Test
-  public void testFailMetricName() {
-
-    assertEquals(null, MetricsNames.failMetricName(null));
-    assertEquals("", MetricsNames.failMetricName(""));
-    assertEquals("tag_fail", MetricsNames.failMetricName("tag"));
-    assertEquals("0_fail", MetricsNames.failMetricName("0"));
-    assertEquals("tag1_fail", MetricsNames.failMetricName("tag1"));
-    assertEquals("tag.tag_fail", MetricsNames.failMetricName("tag.tag"));
-    assertEquals("tag_fail.1", MetricsNames.failMetricName("tag.1"));
-    assertEquals("tag.tag1_fail", MetricsNames.failMetricName("tag.tag1"));
-    assertEquals("tag1.tag2.tag3_fail", MetricsNames.failMetricName("tag1.tag2.tag3"));
-    assertEquals("tag1.tag2_fail.3", MetricsNames.failMetricName("tag1.tag2.3"));
-  }
+    @Test
```

Comment:
Testing of metrics is somewhat limited in the current codebase of Pravega. Given the amount of changes that this PR involves, it would be interesting to think of how to extend the number of tests related to metrics (names, providers, etc.).

Reply:
Yes, adding more tests to meet the diff coverage requirements.